client/packages/common/src/hooks/useDebounce/useDebouncedValueCallback.ts

```diff
@@ -9,10 +9,17 @@ import { FnUtils } from '@common/utils';
 export const useDebouncedValueCallback = <T extends (...args: any[]) => any>(
   callback: T,
   depsArray: DependencyList,
-  wait = 500
+  wait = 500,
+  callbackDepsArray: DependencyList = []
 ): ((...args: Parameters<T>) => Promise<ReturnType<T>>) => {
-  const memoizedCallback = useCallback(FnUtils.debounce(callback, wait), []);
-  const debounced = useCallback(memoizedCallback, depsArray);
+  const memoizedCallback = useCallback(
+    FnUtils.debounce(callback, wait),
+    callbackDepsArray
```

Comment:
adding in dependencies for this callback - otherwise the function is memoized, and in our case, creates a captured enclosure of the `draftOutboundLines` which then gives a 'no stock available' error when allocating

Reply:
I wonder if you would ever need to differentiate between the depsArray and the callbackDepsArray? ie would you  ever need to refresh debounced while leaving the momoizedCallback stale? And if not, is it better to consolidate the two?

Reply:
yes, I do need the differentiation. The original use-case for this hook required that the memoized function be static, while the second `useCallback` is updated based on the data which was to be provided to the first.